### ISSUE TITLE
Fails parsing some link headers

### ISSUE THREAD

**Issue by mal** [peterhellberg/link#2]
Link header from: http://www.w3.org/wiki/LinkHeader

```
Link: <//www.w3.org/wiki/LinkHeader>; rel="original latest-version",<//www.w3.org/wiki/Special:TimeGate/LinkHeader>; rel="timegate",<//www.w3.org/wiki/Special:TimeMap/LinkHeader>; rel="timemap"; type="application/link-format"; from="Mon, 03 Sep 2007 14:52:48 GMT"; until="Tue, 16 Jun 2015 22:59:23 GMT",<//www.w3.org/wiki/index.php?title=LinkHeader&oldid=10152>; rel="first memento"; datetime="Mon, 03 Sep 2007 14:52:48 GMT",<//www.w3.org/wiki/index.php?title=LinkHeader&oldid=84697>; rel="last memento"; datetime="Tue, 16 Jun 2015 22:59:23 GMT"
```

```
runtime error: index out of range
github.com/peterhellberg/link.Parse(0xc82015a000, 0x21b, 0x1)
        /go/src/github.com/peterhellberg/link/link.go:83 +0x66c
github.com/peterhellberg/link.ParseHeader(0xc8200f4840, 0x0)
        /go/src/github.com/peterhellberg/link/link.go:54 +0xd0
github.com/peterhellberg/link.ParseResponse(0xc82011c1b0, 0x4a3959)
        /go/src/github.com/peterhellberg/link/link.go:48 +0x3a
main.main.func3(0xc82011c1b0, 0xc820110050, 0xc8200cb901)
        /go/src/app/main.go:42 +0x28
```

I think this being caused by the commas in datetime values.

http://play.golang.org/p/LX7drOkmNK

**Comment by peterhellberg** [comment]
Yes, this is a bug related to the comman in the from/until values. I’m not sure how to solve this easily since the Go `regexp` package doesn’t support backreferences for performance reasons.

**Comment by peterhellberg** [comment]
@mal Is this fix sufficient for your use case?

**Comment by mal** [comment]
First, thanks for the fix! Unfortunately it won't work for my case. I needed to parse URLs containing commas out of the attributes of the Link header (`Link: <url1>; ref="url,,2,0,4"`). The current solution would end up stripping the commas out of thus invalidating the URL `url204`.

Don't worry though, I found [tent/http-link-go](//github.com/tent/http-link-go) which is a better fit for my particular use case since it uses a tokenizer, so I can also stop parsing as soon as I find the specific Link header I'm looking for.
